app/assets/javascripts/app/form-field-format.js

```diff
@@ -35,6 +35,17 @@ function formatForm() {
       if (el !== '.mfa') document.activeElement.blur();
     }
   });
+
+  const submitBtnSel = 'input[type="submit"]';
+  const btnEl = document.querySelector(submitBtnSel);
+  const safeSubmit = function() {
+    btnEl.disabled = true;
+    document.querySelector('form').submit();
```

Comment:
Doesn't this presume there's only the one form on the page?

I think we should scope this to certain forms, perhaps by a data attribute. Within those forms -- which I think we need to get by queryselectorall, since I think there could be more than one -- I think you want to look for just [type=submit] to capture both button and input. Then from the btnEl onclick, you could look for the correct parent form with btnEl.closest('form') or something like that.

Reply:
I marked it WIP

Reply:
Yes my preferred strategy is for the buttons to opt in (and obviously pick their associated form).  Shocked code climate complaining about >25 lines.  Seems arbitrarily low no?